src/Streamly/Coreutils/Stat.hs

```diff
@@ -12,27 +12,21 @@
 module Streamly.Coreutils.Stat
     ( Stat
     , stat
-    , deRef
-    , file
+    , followLinks
     ) where
 
+import Streamly.Coreutils.Common (Switch(..))
 import System.Posix.Files (FileStatus)
-import qualified System.Posix.Files as Files
-import Data.Default.Class (Default (def))
-
-data Stat = DeRef | File
 
-deRef :: Stat
-deRef = DeRef
+import qualified System.Posix.Files as Files
 
-file :: Stat
-file = File
+newtype  Stat = Stat {deRef :: Switch}
```

Comment:
There is no `Default` instance. How are we going to get the initial value of `Stat` type? We are not exporting the constructor, so there is no way to create it.

We can avoid such problems if we write a simple test, because it will force us to use it.